pip/pep425tags.py

```diff
@@ -198,7 +247,8 @@ def get_supported(versions=None, noarch=False):
                 supported.append(('%s%s' % (impl, versions[0]), abi, arch))
 
         # Has binaries, does not use the Python API:
-        supported.append(('py%s' % (versions[0][0]), 'none', arch))
+        for arch in arches:
+            supported.append(('py%s' % (versions[0][0]), 'none', arch))
```

Comment:
I think ~2000 should not cause any performance issue for pip but it would be worth auditing the codebase to check that this list of tags is not used by algorithm with quadratic complexity or more.